server/src/xml/storage.rs

```diff
@@ -72,15 +72,17 @@ impl Storage {
     }
 }
 
-// @todo #17:35min Implement #xml function in Storage.
-//  This function should return full XML storage has at the moment. #xml
-//  function should be thread-safe, as it intended to be used concurrently.
-//  Don't forget to create a unit tests related to #xml function.
 impl Storage {
-    #[allow(dead_code)]
-    /// Return XML from the storage.
-    pub fn xml() -> Result<()> {
-        Ok(())
+    /// Returns full XML from the storage.
+    // @todo #75:60min Make xml() thread-safe.
+    //  We should make this function thread-safe in order to get sequential of
+    //  reads and write to the store. Don't forget to create a unit-test that
+    //  checks concurrency cases.
+    pub fn xml(self) -> io::Result<String> {
+        let mut file = File::open(self.path)?;
```

Comment:
@h1alexbel let's deprecate it or create a puzzle to avoid `?`

```suggestion
        let mut file = File::open(self.path);
```

Reply:
@l3r8yJ Got it. I found the following problem statements with `?` operator:

* `?` means that the compiler can't use the question-mark to infer that the type you're collecting to is a `Result`.
* `?` calls `into`, which performs type conversion that leads to misleading result types.

Got this from this Stack Overflow  answer: https://stackoverflow.com/a/67644988/19147117

Other sources:
[What is this question mark operator about?](https://stackoverflow.com/questions/42917566/what-is-this-question-mark-operator-about)

Reply:
@l3r8yJ should we stay with `?` in tests?

Reply:
@h1alexbel in tests it's fine